src/bit-systems/quack.ts

```diff
@@ -0,0 +1,20 @@
+import { HubsWorld } from "../app";
+import { Held, Quack } from "../bit-components";
+import { defineQuery, enterQuery } from "bitecs";
+import { SOUND_QUACK, SOUND_SPECIAL_QUACK } from "../systems/sound-effects-system";
+
+const heldQuackQuery = defineQuery([Quack, Held]);
+const heldQuackEnterQuery = enterQuery(heldQuackQuery);
+
+const quack = (eid: number) => {
+  const rand = Math.random();
+  if (rand < Quack.specialQuackPercentage[eid]) {
+    AFRAME.scenes[0].systems["hubs-systems"].soundEffectsSystem.playSoundOneShot(SOUND_SPECIAL_QUACK);
+  } else if (rand < Quack.quackPercentage[eid]) {
+    AFRAME.scenes[0].systems["hubs-systems"].soundEffectsSystem.playSoundOneShot(SOUND_QUACK);
```

Comment:
This was the approach to playing a OneShot used in the camera tool bit system. It feels like this shouldn't be the approach but I didn't see another options without making a bit sound effects system. I might've missed something.

Reply:
That's the right way to do it ATM. It would be good though to reduce the number of AFRAME references as we migrate to bitECS. We hold a AFRAME scene ref in `APP.scene` so you can also get the system through that.